radix-engine-common/src/data/scrypto/custom_schema.rs

```diff
@@ -8,20 +9,22 @@ pub type ScryptoSchema = Schema<ScryptoCustomTypeExtension>;
 pub type ScryptoTypeData<L> = TypeData<ScryptoCustomTypeKind, L>;
 
 /// A schema for the values that a codec can decode / views as valid
-#[derive(Debug, Clone, PartialEq, Eq, Sbor)]
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor, ManifestSbor)]
 pub enum ScryptoCustomTypeKind {
     Reference, /* any */
     GlobalAddress,
     LocalAddress,
     PackageAddress,
     ComponentAddress,
     ResourceAddress,
+    ObjectRef(PackageAddress, String),
```

Comment:
Like the renaming suggestion. Will apply.

> Out of interest, why did we feel it was right to put these on TypeKind rather than on TypeValidation?
> We could have TypeKind of Reference / Own, and then more detailed / constraints on the TypeValidation?

Mainly because I didn't want to maintain two places of stateful validation (inside the traverser and inside the payload validator). 

To some extent, would prefer to keep `TypedTraverser` as stateless as possible because of intended generic use.